torch/fx/experimental/proxy_tensor.py

```diff
@@ -168,108 +310,144 @@ def extract_val(val):
     elif isinstance(val, (int, float, bool)):
         return val
 
+    # BUG? Should this return val?
+    return None
```

Comment:
Is this a valid return type here? Seems like we should using `typing_extensions.assert_never(val)`, right?

Reply:
I kept the existing behavior (returning None. but I made it explicit) because I didn't know what the original code wanted for this case. If you think it should be an assert_never() I'd be happy to make that change.

Reply:
Oh - I didn't realize that `assert_never` turns it into an exhaustive check - that's nice.